action/protocol/managers.go

```diff
@@ -24,6 +24,14 @@ func KeyOption(key []byte) StateOption {
 	}
 }
 
+// KeysOption sets the key for call
+func KeysOption(f func() ([][]byte, error)) StateOption {
```

Comment:
can use `func KeysOption(keys [][]byte) StateOption`?

when `KeysOption` is called, the caller will provide/prepare the keys, if there's an error, caller can handle it on their stack

Reply:
fix this?